### PR TITLE
Add toString, hashCode and equals mapping to Objective-C

### PR DESCRIPTION
**Pull request by hfhbd**
https://youtrack.jetbrains.com/issue/KT-52579/ObjectiveC-Export-toString-hashCode-equals-from-Any-in-KotlinBase#focus=Comments-27-6125427.0-0

### CODE REVIEW

**Review by koshachy — APPROVED** [review]
Thanks for pointing this out!

**Comment by hfhbd** [comment]
Wait, description and hash are functions

**Comment by koshachy** [comment]
> Wait, description and hash are functions, and isEquals too
Yep, but AFAIK they don't use parentheses for signature, do they?

File: docs/topics/native/native-objc-interop.md [hunk 1]
```diff
@@ -64,6 +64,8 @@ same name but different package in the same framework. This algorithm is not sta
 releases.
 As a workaround, you can rename the conflicting Kotlin classes in the framework.
 
+The methods of `kotlin.Any` (`equals()`, `hashCode()` and `toString()`) are mapped to `isEquals(_:)`, `description` and `hash` in Objective-C.
```
[the remark this answers is not in the document]
Comment (hfhbd) [reply docs/topics/native/native-objc-interop.md:67]: Yeah, what about adding the Objective-C as well as Swift names?

File: docs/topics/native/native-objc-interop.md [hunk 2]
```diff
@@ -113,6 +113,10 @@ player.moveTo(LEFT, byMeters = 17)
 player.moveTo(UP, byInches = 42)
 ```
 
+The methods of `kotlin.Any` (`equals()`, `hashCode()` and `toString()`) are mapped
```
Comment (SvyatoslavScherbina) [inline docs/topics/native/native-objc-interop.md:116]: Method order doesn't match the order of Swift/Obj-C names :)

**Review by SvyatoslavScherbina — APPROVED** [review]

File: docs/topics/native/native-objc-interop.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (SvyatoslavScherbina) [reply docs/topics/native/native-objc-interop.md]: Thanks, exactly!

**Review by SvyatoslavScherbina — APPROVED** [review]

**Comment by koshachy** [comment]
Thanks for your contribution! :)

**Pull request merged**